bitnami/thanos/templates/storegateway/tls-server-secret.yaml

```
@@ -0,0 +1,13 @@
+{{- if and (.Values.storegateway.grpc.tls.enabled) (not .Values.storegateway.grpc.tls.existingSecret) }}
```

You don't need parenthesis around the 1st element. Furthermore, you should also check if storegateway is enabled:
```suggestion
{{- if and .Values.storegateway.enabled .Values.storegateway.grpc.tls.enabled (not .Values.storegateway.grpc.tls.existingSecret) }}
```